hs_geo_raster_resource/migrations/custom_migration_for_raster_ori_cov_update_20161107.py

```diff
@@ -0,0 +1,105 @@
+from __future__ import unicode_literals
+
+import os
+import shutil
+import logging
+import tempfile
+import subprocess
+import xml.etree.ElementTree as ET
+
+from django.db import migrations
+from django.core.files.uploadedfile import UploadedFile
+
+from django_irods.storage import IrodsStorage
+from hs_core import hydroshare
+from hs_core.hydroshare.utils import resource_modified, get_file_from_irods
+from hs_geo_raster_resource.models import RasterResource
+from hs_geo_raster_resource import raster_meta_extract
+
+
+def migrate_tif_file(apps, schema_editor):
+    log = logging.getLogger()
+
+    copy_res_fail = []
+    meta_update_fail = []
+    meta_update_success = []
+
+    # start migration for each raster resource that has raster files
+    for res in RasterResource.objects.all():
+        if res.files.all():
+            # copy all the resource files to temp dir
+            temp_dir = ''
+            res_file_tmp_path = ''
+
+            try:
+                temp_dir = tempfile.mkdtemp()
+                for res_file in res.files.all():
+                    res_file_tmp_path = get_file_from_irods(res_file)
+                    shutil.copy(res_file_tmp_path,
+                                os.path.join(temp_dir, os.path.basename(res_file_tmp_path)))
+                    shutil.rmtree(os.path.dirname(res_file_tmp_path))
+
+                vrt_file_path = [os.path.join(temp_dir, f) for f in os.listdir(temp_dir) if '.vrt' == f[-4:]].pop()
+
+            except Exception as e:
+                if os.path.isdir(temp_dir):
+                    shutil.rmtree(temp_dir)
+                if os.path.isfile(res_file_tmp_path):
+                    shutil.rmtree(os.path.dirname(res_file_tmp_path))
+                log.exception(e.message)
+                copy_res_fail.append('{}:{}'.format(res.short_id, res.metadata.title.value))
+                continue
+
+            # update the metadata for the original coverage information of all the raster resources
+            try:
+                meta_updated = False
+
+                # extract meta
+                ori_dir = os.getcwd()
```

Comment:
I may not understand the subtleties here, but can we adopt a model where we limit our use of vrt to where we always use relative path references to files in the same folder rather than the full generality of vrt that can point anywhere.  If we do this, then folders of vrt files can be moved around safely.